export.go

```diff
@@ -1,8 +1,32 @@
 package imageref
 
-type WebpExportParams struct {
-	Quality        int
-	Lossless       bool
-	StripMetadata  bool
+import "github.com/davidbyttow/govips/v2/vips"
+
+type ExportParams struct {
```

Comment:
this doesnt seems like a good idea to me 😅 not very single responsibility-ish

Reply:
- i think the caller should decide which export method to use, not the library
- youre duplicating all possible export params, so duplication is still there, also a little copy&paste is better than a little dependency imo
- as a client i could call it like this `im.ExportWithParams("tif", imageref.ExportParams{})` and would get a png back, which is a pretty bad side effect
- there are no sensible defaults for any params and you cannot create any, as ones that would considered default for webp could be different for png, additionally to call this in multiple places you still need to copy all default parameters
- i would just copy this method to where it is actually called